kubernetes/resource_kubernetes_secret.go

```diff
@@ -38,6 +38,13 @@ func resourceKubernetesSecret() *schema.Resource {
 				Sensitive:   true,
 				Description: "A map of the secret data in base64 encoding. Use this for binary data.",
 			},
+			"base64_data": {
```

Comment:
We recently added a [binary_data](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/kubernetes/resource_kubernetes_secret.go#L35) field to the Secret resource.

```suggestion
			"binary_data": {
```

Reply:
Thank you for the review. This is what I initially tried, but it doesn't help in my use-case. In our situation, the secret is not created by terraform, so we need datasource to be able to read the values of `secret.Data`. Unless I'm misunderstanding this completely, resource's `read` will only populate it if `if v, ok := d.GetOk("binary_data"); ok {` will evaluate to true, which in this case it won't? Also, what `binary_data` does is encode complete bytemap as base64, whereas I really want my keys to remain plaintext, while only encoding (potentially) binary values.

Reply:
To elaborate further - certmanager stores TLS cert as a secret with 3 keys: `tls.crt`, `tls.key` and `keystore.p12`. The former 2 are plain-text, the latter is binary PFX. I wanna fetch that secret retaining the validity of binary value of 'keystore.p12` and still be able to retrieve it even without decoding, passing the value onward to a KeyVault resource payload.

Reply:
Or nevemind my remark about encoding complete bytemap in case of binary_data, I'm sorry, I should've read the code better. But how can I make sure tat datasource populates binary_data?

Reply:
This is great that you've found a coverage gap and come up with a workflow to cover it! I agree, this is not very intuitive. It should probably go into the docs. So here's what I'm thinking for next steps:

1. Let's add `binary_data` attribute in the data source.
2. Let's remove the `base64_data` attribute from the data source and resource.
3. Document how to pull base64 encoded data from the data source (like the example you gave above).
4. Add a test for the use case (copying from the [config_map data source](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/kubernetes/data_source_kubernetes_config_map_test.go#L38) could be helpful, since it has binary_data too).

That can be done in this PR, or I can add the docs and test if you prefer. Thanks,

Reply:
Sounds like a plan. I'll push the additions in a few.